src/adapters/rpc-server/services/upsert-friendship.ts

```diff
@@ -33,20 +36,9 @@ export function upsertFriendshipService({
     logger.debug(`upsert friendship > `, parsedRequest as Record<string, string>)
 
     try {
-      const friendship = await db.getFriendship([context.address, parsedRequest.user!])
-      let lastAction = undefined
-      if (friendship) {
-        const lastRecordedAction = await db.getLastFriendshipAction(friendship.id)
-        lastAction = lastRecordedAction
-      }
+      const lastAction = await db.getLastFriendshipActionByUsers(context.address, parsedRequest.user!)
 
-      if (
-        !validateNewFriendshipAction(
-          context.address,
-          { action: parsedRequest.action, user: parsedRequest.user },
-          lastAction
-        )
-      ) {
+      if (!validateNewFriendshipAction(context.address, parsedRequest, lastAction)) {
```

Comment:
Shouldn't this function also receive the new action the user is trying?

Reply:
It is inside the parsed request